src/eva/vision/models/modules/semantic_segmentation.py

```diff
@@ -22,7 +22,7 @@ class SemanticSegmentationModule(module.ModelModule):
 
     def __init__(
         self,
-        decoder: decoders.Decoder,
+        decoder: decoders.Decoder | nn.Module,
```

Comment:
idk if this is a good idea, as the decoder specifically wants the features to be 4D and the output is also very specific, which makes the Module a bit to loose and generic in my view

Reply:
Good point, but it allows people to use their own decoders without subclassing eva's `Decoder` class.